### DISCUSSION TITLE
Request for Enhanced Software Upgrade Functionality in NetBox

### DISCUSSION THREAD

**Discussion by digital-fortress** [netbox-community/netbox#12981] [Ideas]
Currently, the platform component in NetBox lacks practical capabilities for managing software upgrades, and I believe implementing a feature similar to Nautobot's Device Lifecycle would be great.

The ability to define software versions and track them within NetBox.
Support for associating software versions with devices and managing their upgrade status.
A user-friendly interface for initiating and tracking software upgrade tasks.
Notifications or alerts to inform users about available software updates for their devices.

Please consider.

**Comment by jeremystretch** [comment]
@jinu18 This isn't something that would make sense to build into NetBox, but you're welcome to propose your idea on our [plugin ideas board](https://plugin-ideas.netbox.dev/).
